src/Microsoft.Identity.Web/TokenCacheProviders/MsalAbstractTokenCacheProvider.cs

```diff
@@ -13,6 +20,28 @@ namespace Microsoft.Identity.Web.TokenCacheProviders
     /// <seealso cref="Microsoft.Identity.Web.TokenCacheProviders.IMsalTokenCacheProvider" />
     public abstract class MsalAbstractTokenCacheProvider : IMsalTokenCacheProvider
     {
+        private readonly IServiceProvider? _serviceProvider;
```

Comment:
does the service provider actually need to be a class field or can it be used just in the constructor?

Reply:
correct. this was still in draft and was needed from an earlier implementation. but good catch and thank you!